mirrord/layer/src/socket/ops.rs

```diff
@@ -226,7 +245,10 @@ pub(super) fn bind(
 
     SOCKETS.insert(sockfd, socket);
 
-    Detour::Success(bind_result)
+    // node reads errno to check if bind was successful and doesn't care about the return value
+    // (???)
+    errno::set_errno(errno::Errno(0));
```

Comment:
Nitpick: would be nicer to get the errno before we call `bind_port` and restore the original errno.

Reply:
afaik `bind_port` doesn't reset the errno and that's why we got there in the first place

Reply:
I'm not sure I understand what you're saying, but bind's supposed to only change the errno if there is an error, right?
Because now we also change it (to 0) if everything's fine, which is different than bind's behaviour, no?

If for example a user application decides to do:

1. set errno to 0.
2. do first operation that might fail and set errno.
3. bind port.
4. check errno to see if 2 or 3 failed.

In 3 we would set errno to 0 and erase the errno from 2 and the user application would lose that information.

I don't think that's a big problem, still more of a nitpick, but I want to make sure we understand each other 😃

Reply:
I know that, but Node checks errno even if success code is 0 so we have to reset it. I also mentioned in the comment explaining the above...

Reply:
Yeah, but we can reset it to the value it had before we did our detour.

Reply:
get errno before we do anything that can change errno, then restore that value when we're done.

Reply:
if it was 0 before, it's 0 after we're done.

Reply:
mm, ok not bad idea but we merged it already G: